### PR TITLE
Better documentation for jnp.load

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -320,11 +321,43 @@ def _convert_and_clip_integer(val: ArrayLike, dtype: DType) -> Array:
   return clip(val, min_val, max_val).astype(dtype)
 
 
-@util.implements(np.load, update_doc=False)
-def load(*args: Any, **kwargs: Any) -> Array:
+def load(file: IO[bytes] | str | os.PathLike[Any], *args: Any, **kwargs: Any) -> Array:
```
Comment (superbobry) [inline jax/_src/numpy/lax_numpy.py:324]: OOC can we use `os.PathLike[str]` here or do we want to allow both `str` and `bytes`?

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -320,11 +321,43 @@ def _convert_and_clip_integer(val: ArrayLike, dtype: DType) -> Array:
...
+  """Load JAX arrays from npy files.
+
+  JAX wrapper of :func:`numpy.load`.
+
+  This function is a simple wrapper of :func:`numpy.load`, but in the case of
+  ``.npy`` files created with :func:`numpy.save` or :func:`jax.numpy.save`,
+  the output will be returned as a :class:`jax.Array`, and ``bfloat16`` data
+  types will be restored. For ``.npz`` files, results will be returned as
```
Comment (superbobry) [inline jax/_src/numpy/lax_numpy.py:332]: I suspect it should be possible to return `jax.Array`s for `.npz` files as well. Is there a reason why we don't do that?

**Review by superbobry — APPROVED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:332]: loading an `npz` returns a lazy view of the file buffer, with various methods to actually load the (NumPy) arrays. Having it return JAX arrays would require re-implementing that object.

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:324]: I'm honestly not sure, I thought `PathLike[Any]` would be the safest annotation here.

**Pull request merged**